app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt

```diff
@@ -0,0 +1,130 @@
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchWidget(
+    text: String,
+    onTextChanged: (String) -> Unit,
+    onSearchClicked: (String) -> Unit,
+    onCloseClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+
+    Surface(
+        modifier = modifier
+            .fillMaxWidth()
+            .height(TOP_BAR_HEIGHT),
+        shadowElevation = AppBarDefaults.TopAppBarElevation,
+        color = MaterialTheme.colorScheme.primary
+    ) {
+        TextField(
+            modifier = Modifier
+                .fillMaxWidth(),
+            value = text,
+            onValueChange = {
+                onTextChanged(it)
+            },
+            placeholder = {
+                Text(
+                    text = stringResource(R.string.search_placeholder),
+                    modifier = Modifier.alpha(ContentAlpha.medium),
+                    color = MaterialTheme.colorScheme.onPrimary,
+                    style = MaterialTheme.typography.bodyLarge
+                )
+            },
+            textStyle = MaterialTheme.typography.bodyLarge.copy(
+                color = MaterialTheme.colorScheme.onPrimary
+            ),
+            singleLine = true,
+            leadingIcon = {
+                IconButton(
+                    modifier = Modifier
+                        .alpha(ContentAlpha.medium),
+                    onClick = { }
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.Search,
+                        contentDescription = stringResource(R.string.search_icon),
+                        tint = MaterialTheme.colorScheme.onPrimary
+
+                    )
+                }
+            },
+            trailingIcon = {
+                IconButton(
+                    onClick = {
+                        if (text.isNotEmpty()) {
+                            onTextChanged("")
+                        } else {
+                            onCloseClicked()
+                        }
+                    }
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.Close,
+                        contentDescription = stringResource(R.string.close_icon),
+                        tint = MaterialTheme.colorScheme.onPrimary
+
+                    )
+                }
+            },
+            keyboardOptions = KeyboardOptions(
+                imeAction = ImeAction.Search
+            ),
+            keyboardActions = KeyboardActions(
+                onSearch = {
+                    onSearchClicked(text)
+                    keyboardController?.hide()
+                }
+            ),
+            colors = TextFieldDefaults.textFieldColors(
+                containerColor = Color.Transparent,
+                cursorColor = MaterialTheme.colorScheme.onPrimary
+            )
+        )
+    }
+}
+
+@Preview(name = "Light")
```

Comment:
Thank you for the previews, but like the other comment they don't seem to do light/dark modes.

Reply:
Morning, I did some experimenting and found a way of displaying the previews correctly. I will implement the change on both PR's.